ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c

```diff
@@ -514,6 +517,43 @@ nnfw_getOutputDim (const GstTensorFilterProperties * prop,
   return 0;
 }
 
+/**
+ * @brief Internal function to invoke with dummy data.
+ * When changing the input shape, NNFW will update the output shape after the invoke process is done.
+ */
+static void
+nnfw_invoke_dummy (const GstTensorFilterProperties * prop, void **private_data,
+    GstTensorsInfo * in_info, GstTensorsInfo * out_info)
+{
+  GstTensorMemory input[NNS_TENSOR_SIZE_LIMIT] = { 0, };
+  GstTensorMemory output[NNS_TENSOR_SIZE_LIMIT] = { 0, };
+  guint i;
+
+  for (i = 0; i < in_info->num_tensors; ++i) {
+    input[i].size = gst_tensor_info_get_size (&in_info->info[i]);
+    input[i].data = g_malloc (input[i].size);
+  }
+
+  /* The output shape would be changed, set enough size for output buffer. */
+  for (i = 0; i < out_info->num_tensors; ++i) {
+    output[i].size = gst_tensor_info_get_size (&out_info->info[i]) * 2;
+    output[i].data = g_malloc (output[i].size);
+  }
+
+  while (nnfw_invoke (prop, private_data,  input, output) != 0) {
+    g_warning ("Invoke failed, reallocate output tensors and retry.");
+    for (i = 0; i < out_info->num_tensors; ++i) {
+      output[i].size *= 2;
+      output[i].data = g_realloc (output[i].data, output[i].size);
+    }
+  }
+
+  for (i = 0; i < in_info->num_tensors; ++i)
+    g_free (input[i].data);
+  for (i = 0; i < out_info->num_tensors; ++i)
+    g_free (output[i].data);
```

Comment:
How about setting buffers to nullptr to prevent any possible side-effects?

Reply:
updated